File: Deepseek-Coder-33b-Instruct/stage2/declare_directive_create_clause.c
#include <stdlib.h>
#include <openacc.h>

#ifndef T1
//T1:declare directive create clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    int *host_data = (int*)malloc(sizeof(int));
    *host_data = 10;

    #pragma acc data create(host_data[0:1])
    {
        host_data[0] = 20;
    }

    if (*host_data != 20) {
        err = 1;
    }

    free(host_data);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/declare_directive_create_clause.f90
!T1:declare directive create clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        REAL(KIND=8), DIMENSION(:), ALLOCATABLE :: a
        INTEGER :: i

        ALLOCATE(a(N))

        !$acc data create(a)
        !$acc parallel loop
        DO i = 1, N
          a(i) = i
        END DO
        !$acc end parallel loop
        !$acc end data

        ! Check if the array is allocated correctly
        DO i = 1, N
          IF (a(i) .ne. i) THEN
            errors = errors + 1
          END IF
        END DO

        DEALLOCATE(a)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM